lib/licensee/projects/github_project.rb

```diff
@@ -0,0 +1,43 @@
+# Github project
+#
+# Analyses a remote Github repository for license information
+#
+# Only the root folder of a repository will be scanned for performance reasons!
+
+require 'octokit'
+
+module Licensee
+  module Projects
+    class GithubProject < Licensee::Projects::Project
+      # If there's any trailing data (e.g. `.git`) this pattern will ignore it:
+      # we're going to use the API rather than clone the repo.
+      GITHUB_REPO_PATTERN = %r{https://github.com/([^\/]+\/[^\/\.]+).*}
+
+      def initialize(github_url, **args)
+        @repo_name = github_url[GITHUB_REPO_PATTERN, 1]
+        raise ArgumentError, "Not a github URL: #{github_url}" unless @repo_name
+        super(**args)
+      end
+
+      private
+
+      def files
+        @files ||= fetch_files.map do |data|
+          {
+            name:         data[:name],
+            contents_url: data[:download_url],
+            dir:          '/'
+          }
+        end
+      end
+
+      def load_file(file)
+        Octokit.get URI(file[:contents_url])
```

Comment:
I believe we can use `Octokit.contents @repo_name, path: file[:name]` here to rely on Octokit internals a bit more (and save us from needing to store the `contents_url` in the hash).

Reply:
Is there any reason we'd need to memoize/cache this method? I don't think so (and if so, could be a good future improvement).